tests/gds_kernel_latency.c

```diff
@@ -331,6 +338,22 @@ static struct pingpong_context *pp_init_ctx(struct ibv_device *ib_dev, int size,
 			fprintf(stderr, "Failed to modify QP to INIT\n");
 			goto clean_qp;
 		}
+	} else {
+		struct ibv_qp_attr attr = {
+			.qp_state        = IBV_QPS_INIT,
+			.pkey_index      = 0,
+			.port_num        = port,
+			.qp_access_flags = IBV_ACCESS_REMOTE_WRITE | IBV_ACCESS_REMOTE_READ | IBV_ACCESS_LOCAL_WRITE
+		};
+
+		if (ibv_modify_qp(ctx->qp, &attr,
```

Comment:
can't you consolidate the call to ibv_modify_qp() into a single one after the if/else block ?